src/werkzeug/routing.py

```diff
@@ -662,12 +679,23 @@ def __init__(
         self.defaults = defaults
         self.build_only = build_only
         self.alias = alias
-        if methods is None:
-            self.methods = None
-        else:
+        self.websocket = websocket
+        if methods is not None:
             if isinstance(methods, str):
                 raise TypeError("param `methods` should be `Iterable[str]`, not `str`")
-            self.methods = set([x.upper() for x in methods])
+            methods = set([x.upper() for x in methods])
+
+        if websocket and methods is not None and methods != {"GET"}:
```

Comment:
Should this also allow `OPTIONS`? Flask automatically adds the OPTIONS method if a route doesn't specify it, and CORS needs it.

Reply:
Yea, and `HEAD` I think.